macOS nightly build fails due to qt update from homebrew

<!-- For information about contributing see: https://github.com/supercollider/supercollider/wiki/Contributing-directory -->
<!-- Please check there isn't already an issue on this topic. -->

Currently we have some problems regarding Qt6 - this could be due to Qt updates from homebrew?
My local build still somehow succeeds though? (maybe there is a difference in silicon vs intel mac?)

See e.g. a currently failing build https://github.com/supercollider/supercollider/actions/runs/13587575374/job/37985931506#step:13:89 using

```
 ==> Installing qt
🍺  /usr/local/Cellar/qt/6.8.2: 16,372 files, 348.2MB
```

which fails

```
 -- Could NOT find Qt6WebEngineCore (missing: Qt6WebEngineCore_DIR)
CMake Error at QtCollider/CMakeLists.txt:38 (find_package):
  Found package configuration file:

    /usr/local/lib/cmake/Qt6/Qt6Config.cmake

  but it set Qt6_FOUND to FALSE so package "Qt6" is considered to be NOT
  FOUND.  Reason given by package:

  Failed to find required Qt component "WebEngineCore".

  Expected Config file at
  "/usr/local/lib/cmake/Qt6WebEngineCore/Qt6WebEngineCoreConfig.cmake" does
  NOT exist



  Configuring with --debug-find-pkg=Qt6WebEngineCore might reveal details why
  the package was not found.

  Configuring with -DQT_DEBUG_FIND_PACKAGE=ON will print the values of some
  of the path variables that find_package uses to try and find the package.

Call Stack (most recent call first):
  lang/CMakeLists.txt:184 (include)
```

The last build from dev which still worked was https://github.com/supercollider/supercollider/actions/runs/13252984681/job/36994687033 which was using

```
 ==> Installing qt
==> Caveats
You can add Homebrew's Qt to QtCreator's "Qt Versions" in:
  Preferences > Qt Versions > Link with Qt...
pressing "Choose..." and selecting as the Qt installation path:
  /usr/local
==> Summary
🍺  /usr/local/Cellar/qt/6.7.3: 15,091 files, 679.6MB
```

See https://github.com/Homebrew/homebrew-core/commits/555cc061aa19066b0e0428e8858c231de0c4f70a/Formula/q/qt.rb for diff in the homebrew formula.

There is also 300 mbs in diff between both versions which seems suspicious.

--- a/.github/workflows/build_macos.yml
+++ b/.github/workflows/build_macos.yml
@@ -55,11 +55,11 @@ jobs:
             extra-cmake-flags: ""
             artifact-suffix: "macOS-x64-legacy" # set if needed - will trigger artifact upload
 
-          - job-name: "x64 use system libraries"
-            os-version: "13"
-            xcode-version: "15.2"
+          - job-name: "arm64 use system libraries"
+            os-version: "15"
+            xcode-version: "16.0"
             deployment-target: ""
-            cmake-architectures: x86_64
+            cmake-architectures: "arm64"
             homebrew-packages: "qt@6 libsndfile readline fftw portaudio yaml-cpp boost"
             vcpkg-packages: ""
             vcpkg-triplet: ""

--- a/.github/workflows/build_macos.yml
+++ b/.github/workflows/build_macos.yml
@@ -68,9 +68,11 @@ jobs:
           - job-name: "x64 shared libscsynth"
             os-version: "13"
             xcode-version: "15.2"
+            qt-version: "6.7.3" # will use qt from aqtinstall
+            qt-modules: 'qtwebengine qtwebchannel qtwebsockets qtpositioning'
             deployment-target: ""
             cmake-architectures: x86_64
-            homebrew-packages: "qt@6 libsndfile readline fftw portaudio"
+            homebrew-packages: "libsndfile readline fftw portaudio"
             vcpkg-packages: ""
             vcpkg-triplet: ""
             extra-cmake-flags: "-D LIBSCSYNTH=ON"

